src/go/rdctl/cmd/snapshot.go

```diff
@@ -113,8 +99,8 @@ func ensureBackendStarted() error {
 }
 
 func ensureBackendStopped(cmd *cobra.Command) error {
-	connectionInfo, err := getConnectionInfo()
-	if err != nil || connectionInfo == nil {
+	connectionInfo, err := config.GetConnectionInfo()
+	if err != nil {
```

Comment:
I believe we want to return `nil` when `errors.Is(err, os.ErrNotExist)`. This case isn't an error - it is expected when the main process is not running.

Same comment for the `GetConnectionInfo()` call in `ensureBackendStarted()`.

Reply:
Yes, but only when the user did not explicitly specify `--config-path`. Otherwise it should still be an error, right?

So maybe we need to return a different error when the default config file doesn't exist? Or maybe we need to return `nil` for the `connectionInfo` in that case (just like we did before), and then this condition remains `err != nil || connectionInfo == nil`.

That seems the right way to go, but lmk wdyt?

Reply:
Yes, you're right, it is a special case when the config file does not exist and the user did not specify `--config-path`. I think the way to handle this would be to return a special error. I had `ErrMainProcessNotRunning` in my PR, though that name isn't great because it is making an assumption. `ErrDefaultConfigFileNotExist`?

I think that returning a `nil` `connectionInfo` would be un-idiomatic and confusing.

Reply:
Oops, already commented on the other comment... Let me know what you think about the `mayBeMissing` option!